_posts/tumblr/2015-04-27-learning-git-with-mario-bros.md

```diff
@@ -0,0 +1,34 @@
+---
+layout:     post
+title:      Learning Git with Mario Bros
+date:       2015-04-27 09:41:29
+summary:    Like playing Mario Bros? Here's one programming lesson hidden in the famous videogame.
+categories: code
+author:     stefy
+---
+![mariobros](/images/mariobros.jpg)
+
+
+###What Mario Bros and GitHub have in common:
+
+Part of learning how to program is understanding how GitHub works. Github is an open source project started by Linux creator Linus Torvalds. It basically allows you to make changes to a project by copying the repository to your own system. Then you can make your changes on your local copy and “check them in” to the central server. When I had trouble putting this to practice, Max, our awesome, back-end lead, came up with this brilliant, yet simple, explanation:
+
+Max began explaining that a **commit** is just like a "checkpoint": a snapshot of your files in a determined point in time. Now, imagine you are in a Mario game. You're in the middle of world 4.  All you see is where you are right now: you have 50 tokens and 3 lives while you are running for you life! You want to know how many lives and chips you had when you crossed the last "checkpoint" (meaning when you passed from world 3 to 4). 
+
+To do that, you use `git checkout` (hereinafter referred to as `git co`). Now, what `git co` allows you is traveling through time to achieve what you want. Next, you write `git co HEAD~1` (because the last commit was made when changing worlds). Now this does not mean you have to go back through all levels of world 4. Instead, it lets you see what was going on in the past.  Once satisfied with your changes, you write `git co master` and return to where you were. 
```

Comment:
Está un poco confuso este párrafo. Creo que podría ser más claro si especificas qué significa "achieve what you want". What is my problem, How does `git co` solve it? Y después hacer el walk thru step by step